server/lib/src/idm/account.rs

```diff
@@ -794,6 +809,34 @@ impl Account {
         let vap = Value::ApplicationPassword(ap);
         Ok(ModifyList::new_append(Attribute::ApplicationPassword, vap))
     }
+
+    pub(crate) fn to_unixusertoken(&self, ct: Duration) -> Result<UnixUserToken, OperationError> {
+        let (gidnumber, shell, sshkeys, groups) = match &self.unix_extn {
+            Some(ue) => {
+                let sshkeys: Vec<String> = ue.sshkeys.keys().cloned().collect();
+                (ue.gidnumber, ue.shell.clone(), sshkeys, ue.groups.clone())
+            }
+            None => {
+                return Err(OperationError::InvalidAccountState(
+                    "Missing class: posixaccount".to_string(),
```

Comment:
Minor nit around string-typo-avoidance, you'll need to check `EntryClass` is imported.

```suggestion
                    format!("Missing class: {}", EntryClass::PosixAccount),
```